<launch>
  <node pkg="tf" type="static_transform_publisher" name="base_link_broadcaster" args="0 0 0 1.57 3.1416 1.57 map link1 100" />
  
  <node pkg="creative_interactive_gesture_camera" type="main" name="depth_cam" respawn="false"  required="true">
    <remap from="/points2" to="/depth_registered/pointsRGB" /> <!-- output -->
    <remap from="/points_xyz" to="/depth_registered/points" /> <!-- output -->
    <remap from="/rgb_data" to="/rgb/image_raw"/> <!-- output -->
    <remap from="/camera_info" to="/rgb/camera_info" /> <!-- output -->
    <remap from="/mono_data" to="/mono/image_raw" /> <!-- output -->
    <remap from="/camera_mono_info" to="/mono/camera_info" /> <!-- output -->
    <remap from="/cloud_image" to="/depth/points/image_raw" /> <!-- coutput -->
  </node>
  
  <!--<node pkg="creative_interactive_gesture_camera" type="main2" name="pose_estimate" respawn="false" >
    <remap from="/points2" to="/depth_registered/points" /> --> <!-- input -->
 <!-- </node> -->
  
  <node pkg="rviz" type="rviz" name="rviz" args="-d /home/common/catkin_ws/src/creative_interactive_gesture_camera/src/rvizconfig.rviz" respawn="false"  required="true">
  </node>
  
  <node pkg="image_view" type="image_view" name="image_view" args="image:=/rgb/image_raw" respawn="false"  required="true">
  </node> 
  <node pkg="image_view" type="image_view" name="image_view_mono" args="image:=/mono/image_raw" respawn="false"  required="true">
  </node> 
  
  <node pkg="creative_interactive_gesture_camera" type="point_cloud_avg" name="point_cloud_avg" respawn="false"  required="true">
    <remap from="/points2" to="/depth_registered/pointsRGB" /> <!-- input -->
    <remap from="/transform_points2" to="/depth_registered/averaged/pointsRGB" /> <!-- output -->
  </node>
  
  <node pkg="creative_interactive_gesture_camera" type="camera_sync" name="camera_sync" respawn="false"  required="true">
    <remap from="/rgb_image" to="/rgb/image_raw" /> <!-- input -->
    <remap from="/camera_info" to="/rgb/camera_info" /> <!-- input -->
    <remap from="/point_cloud" to="/depth_registered/pointsRGB" /> <!-- input -->
    <remap from="/points2_image" to="/depth_registered/points_imageRGB" /> <!-- output -->
  </node>
  
  <!-- Run compareDepthImage for depth image debugging -->
  <node pkg="creative_interactive_gesture_camera" type="compareDepthImage" name="compareDepthImage" respawn="false"  required="true">
    <remap from="depth_image" to="/depth/points/image_raw" /> <!-- input -->
  </node>
  
  <node pkg="nodelet" type="nodelet" name="rgb_proc_nodelet" 
        args="standalone image_proc/crop_decimate camera:=rgb camera_out:=rgb_downsampled">
  </node>
  
  <node pkg="dynamic_reconfigure" type="dynparam" name="reconfig_x" 
        args="set /rgb_proc_nodelet decimation_x 4">
  </node>
  
  <node pkg="dynamic_reconfigure" type="dynparam" name="reconfig_y" 
        args="set /rgb_proc_nodelet decimation_y 3">
  </node> 
  
  <node pkg="nodelet" type="nodelet" name="mono_proc_nodelet" 
        args="standalone image_proc/crop_decimate camera:=mono camera_out:=mono_downsampled">
  </node>
  
  <node pkg="dynamic_reconfigure" type="dynparam" name="reconfig_monox" 
        args="set /mono_proc_nodelet decimation_x 4">
  </node>
  
  <node pkg="dynamic_reconfigure" type="dynparam" name="reconfig_monoy" 
        args="set /mono_proc_nodelet decimation_y 3">
  </node>
  
  <node pkg="image_view" type="image_view" name="image_view2" args="image:=/rgb_downsampled/image_raw" respawn="false"  required="true">
  </node> 
  
  <node pkg="image_view" type="image_view" name="image_viewmono2" args="image:=/mono_downsampled/image_raw" respawn="false"  required="true">
  </node> 
  
  <node pkg="image_view" type="image_view" name="image_view3" args="image:=/depth_registered/points_imageRGB" respawn="false"  required="true">
  </node> 
  
</launch>
